### ISSUE TITLE
Plugin is silent with Hadolint 1.2.1, and probably needs updating to new hadolint stdout report format

### ISSUE THREAD

**Issue by ronjouch** [niksite/SublimeLinter-contrib-hadolint#1]
SublimeLinter-hadolint never shows any linter warning, even though command-line hadolint works.
- `hadolint` in in my `PATH` and other linters work well
- hadolint --version returns `Haskell Dockerfile Linter v1.2.1`, which still makes `linter.py`'s `version_re` and `version_requirement` happy
- Looking at the log below, I tried to replace the linter's `cmd = 'hadolint -i'` by just `hadolint`, since there seems to be no `-i` flag in my hadolint version (trying it returns `Invalid option '-i'`). Was the `-i` flag to support passing stdin input? If yes and if it was removed, is there a way to tell SublimeLinter to pass the filepath instead?
- I'm not sure what else to try.
- Debug log:

```
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /tmp/SublimeLinter3-ronj 
SublimeLinter: hadolint version query: /home/ronj/.local/bin/hadolint -v 
SublimeLinter: hadolint version: 1.2 
SublimeLinter: hadolint: (>= 0.1) satisfied by 1.2 
SublimeLinter: hadolint activated: /home/ronj/.local/bin/hadolint 
SublimeLinter: hadolint: Dockerfile ['/home/ronj/.local/bin/hadolint'] 
SublimeLinter: hadolint output:
Please provide a Dockerfile
```

---

Additional note: once it works, the `regex` may have to be updated to hadolint's current format, which previously assumed the line number immediately. Currently here's what it looks like:

```
./Dockerfile:1 DL3006 Always tag the version of an image explicitly.
```

, so something like `r'(.+)\:(?P<line>[^\s]+) (?P<message>.+)'` might work now

**Comment by thuandt** [comment]
@ronjouch I pushed my PR with fix for `hadolint v1.2.1`
@niksite can you review my PR?

**Comment by ronjouch** [comment]
@niksite @thuandt fix confirmed with linter-hadolint 1.0.1 / hadolint 1.2.1 on Sublime 3 build 3125. Thanks! Closing.
